### PR TITLE
8.0 Update: gRPC Json Transcoding

### LINKED ISSUE #29080: 8.0: gRPC: http.proto and annotations.proto are now included and auto referenced
In just the 8.0 version of the document, update the Usage section so it no longer instructs to add http.proto and annotations.proto,  Also in step one consider noting those packages are included with the NuGet package.

Do a sweep to look for any other topics where the same update is needed.

Copying http.proto and annotations.proto files into the app is no longer required. As of .NET 8 preview 1 those files are included in the NuGet package and automatically referenced.

See: [gRPC JSON transcoding no longer requires http.proto and annotations.proto(]https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-1/#grpc-json-transcoding-no-longer-requires-http-proto-and-annotations-proto
gRPC JSON transcoding no longer requires http.proto and annotations.proto)

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 25375ca1-0ce8-a3ca-0c3a-446328f09ba1
* Version Independent ID: 25375ca1-0ce8-a3ca-0c3a-446328f09ba1
* Content: [gRPC JSON transcoding in ASP.NET Core gRPC apps](https://learn.microsoft.com/en-us/aspnet/core/grpc/json-transcoding?view=aspnetcore-8.0)
* Content Source: [aspnetcore/grpc/json-transcoding.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/grpc/json-transcoding.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-grpc**
* GitHub Login: @JamesNK
* Microsoft Alias: **jamesnk**

---
[Associated WorkItem - 87021](https://dev.azure.com/msft-skilling/Content/_workitems/edit/87021)

### LINKED ISSUE #29321: 8.0 and 7.0 app samples for Json Transcoding
Move the inline code for the _Usage_ and _Annotate gPRC methods_ sections for JSON Transcoding to an app sample for .NET 7 and a .NET 8 app sample so the differences and the basic set up is clear.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 25375ca1-0ce8-a3ca-0c3a-446328f09ba1
* Version Independent ID: 25375ca1-0ce8-a3ca-0c3a-446328f09ba1
* Content: [gRPC JSON transcoding in ASP.NET Core gRPC apps](https://learn.microsoft.com/en-us/aspnet/core/grpc/json-transcoding?view=aspnetcore-7.0)
* Content Source: [aspnetcore/grpc/json-transcoding.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/grpc/json-transcoding.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-grpc**
* GitHub Login: @JamesNK
* Microsoft Alias: **jamesnk**

---
[Associated WorkItem - 91425](https://dev.azure.com/msft-skilling/Content/_workitems/edit/91425)

### PR DESCRIPTION
**Pull request by wadepickett**
[Use section changes in version 8 - Internal review only](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/json-transcoding?branch=pr-en-us-29279&view=aspnetcore-8.0#usage)

Fixes #29080
Fixes #29321

* Usage section updated to no longer instruct to add http.proto, annotations.proto and note they are included with the transcoding package.
* Added .NET 8 version of a JSON transcoding app sample. 
* Added .NET 7 version of a JSON transcoding app showing old structure for http.proto and annotations.proto
* Moved topic to new versioning by include and updated related paths.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/grpc/json-transcoding.md](https://github.com/dotnet/AspNetCore.Docs/blob/9ef48884f64501a939cbc7b5b627935e3eff63ba/aspnetcore/grpc/json-transcoding.md) | [gRPC JSON transcoding in ASP.NET Core gRPC apps](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/json-transcoding?branch=pr-en-us-29279) |

### CODE REVIEW

**Review by tdykstra — APPROVED** [review]
LGTM

File: aspnetcore/grpc/json-transcoding.md [hunk 1]
```diff
@@ -4,14 +4,16 @@ author: jamesnk
 description: Learn how to create JSON HTTP APIs for gRPC services using gRPC JSON transcoding.
 monikerRange: '>= aspnetcore-7.0'
 ms.author: jamesnk
-ms.date: 02/19/2023
+ms.date: 05/10/2023
 uid: grpc/json-transcoding
 ms.custom: engagement-fy23
 ---
 # gRPC JSON transcoding in ASP.NET Core gRPC apps
```
Comment (JamesNK) [inline aspnetcore/grpc/json-transcoding.md:11]: Unrelated change: shorten the title slightly so it doesn't wrap awkwardly.
```suggestion
# gRPC JSON transcoding in ASP.NET Core
```

**Review by JamesNK — APPROVED** [review]
Looks good to me
